### ISSUE TITLE
Rescan USB from initial startup popup window.

### ISSUE THREAD

**Issue by RUDEWORLD** [bitfocus/companion#515]
**Describe the feature**
Hi, I would like to see the Rescan USB button to the startup window for companion. The popup that you get as soon as you start it, before launching the GUI.

**Is this platform dependent (windows, mac, ..)?**
NO

**If documentation is required to implement, do you know where to find it?**
N/A

**Usecases**
Any time I unplug my stream deck, I have to rescan the USB to get companion to pop up on the buttons. Currently there are multiple pages to navigate to get to the rescan button. Seeing as it the most commonly used button once everything is setup, I would love to see this on the initial startup page, so I can scan it immediately after plugging in without having to navigate through the GUI

**Comment by hadphild** [comment]
There has been talk of having a rescan after .... Minutes of lose of panels. 

Having a tick box as a preference would be good that could force rescan. (Dont know how easy that would be).

**Comment by willosof** [comment]
I think we'll rescan 5 seconds after a disconnect in any case, and rescan all every 1 minute or something. We'll just make sure it doesn't block anything anymore. I think it doesn't, after @haakonnessjoen moved that part of the code out in forks.

**Comment by robinhoodis** [comment]
my stream deck constantly disconnects and then I have to open the gui and find the rescan USB button.

**Comment by arigolan** [comment]
Any progress with this? 
I would love to see a "Rescan USB for devises after Restart" option. 
(I tried adding a button on Stream Deck to Restart companion, have a 10000ms delay, then Rescan USB for devices- but it obviously didn't work. I'm guessing the second command is lost upon the shutdown, so it never knows to rescan upon starting back up...)

**Comment by JeffreyDavidsz** [comment]
When starting companion it scans for devices, so when your deck is not recognized something else is going wrong

**Comment by arigolan** [comment]
For us it is 100% consistent, that when we either restart companion or unplug and replug the Stream Deck USB cable, that it does not come back until we explicitly do a Surfaces: Rescan USB. Sometimes the rescan works fine, other times we need to unplug/plug and rescan. Under the surfaces tab, only the emulator is visible until we rescan. And we get the following log warning:
  
26. 14:03:59 USB(undefined): Found device, but no access. Please quit any other applications using the device, and try again.

Hopefully we don't need to quit ally the other applications- currently we are running an ATEM switcher, an Iris Down Timer Clock, a DigiCo Audio Mixer, and via generic Http & Vicreo, mimoLive...

Any thoughts or things you'd like me to test?

**Comment by JeffreyDavidsz** [comment]
Sounds to me the streamdeck app is also starting at start-up. Sometimes is a bit hard to find it. Look for it and make sure it's closed and not starting again at startup

**Comment by arigolan** [comment]
I've never installed the streamdeck app.. upon recommendation of colleagues, I went straight away to Companion...

**Comment by arigolan** [comment]
I also confirmed streamdeck is not running via Activity Monitor (in case someone unbeknownst to me had installed it, as unlikely as that is)

**Comment by JeffreyDavidsz** [comment]
Did you ever tried it with an other deck? Normally your error is regarding the app or newer deck with older companion version

**Comment by arigolan** [comment]
No, this is the only one we have. I can check the firmware of the Streamdeck XL and get back to you. I'm believe I need the streamdeck software to do that, so I'll install that on a separate computer to keep the normal system clean.

**Comment by JeffreyDavidsz** [comment]
You are using companion > 2.0 right?

**Comment by arigolan** [comment]
Just confirmed the Stream Deck XL is up to date running 4.9.0 (13177)
Companion v2.1.0 (2.1.0-47823d0-2493)

**Comment by JeffreyDavidsz** [comment]
All right let me know it the deck works on a different machine (before you install the app) ;-)

**Comment by arigolan** [comment]
Same behavior running on different machine running different OS (primary Companion server was on Mac OSX 10.11.6 El Capitan on Mac mini, now testing on MacBookPro running 10.14.6 Mojave. Could having Instances that are not running/connected cause this behavior (both machines do)?

**Comment by JeffreyDavidsz** [comment]
It should not, but for error testing you should clear the database of companion on the other machine, test it step by step. By installing several things you cannot rule out where the problem is. It could still be the deck itself. But when you use it on a machine that has the same setup, I cannot tell

**Comment by arigolan** [comment]
After updating OS on mac mini to High Sierra, it seems to behave a lot better, including a weird issues where osc/path field would not retain data after navigating away from button and back. If problem persists, I'll let you know.

**Comment by josephdadams** [comment]
@Julusian should we close this now that we have the context menu option?
